### PR TITLE
fix(ui5-tabcontainer): adding selected tab programatically deselects previously selected tab

### PR DESCRIPTION
**Pull request by kskondov**
Fixes: #5932

### CODE REVIEW

**Review by gmkv — CHANGES_REQUESTED** [review]
I don't think we should change the state of the tabs which are created by the application.
```
selectedTabs.forEach(item => {
	if (item !== this._selectedTab) {
		item.selected = false;
		item._selected = false;
	}
});
```
Maintaining only one selected tab in the slot should be a responsibility of the application as well.

Try to replicate the exact case that is reported in the issue, maybe the bug is not exactly that.

File: packages/main/test/pages/TabContainer.html [hunk 1]
```diff
@@ -796,6 +796,14 @@ <h2>Text only Start And End Overflow Custom Overflow Buttons</h2>
 			<ui5-tabcontainer id="tabContainerEmpty">
 			</ui5-tabcontainer>
 		</section>
+		<section>
+			<h3>Add new selected tab programmatically</h3>
+			<ui5-tabcontainer id="tabContainerAddTabsProgrammatically">
+				<ui5-tab text="Tab 0"></ui5-tab>
```
Comment (gmkv) [inline packages/main/test/pages/TabContainer.html:802]: Make at one of those tabs selected, so that the code which resets the selection on line 883 will work - to replicate the real case. In the test, test the state of the tabs before the "Act".

**Review by gmkv — CHANGES_REQUESTED** [review]
Rewrite the title of the PR or add a message to explain what is the edge case is being fixed. "Only one tab can be selected" will still not be true after this commit.

**Review by gmkv — APPROVED** [review]
